backend/src/main/java/org/eclipse/tractusx/puris/backend/stock/logic/adapter/ItemStockSammMapper.java

```diff
@@ -0,0 +1,277 @@
+        samm.setPositions(new ArrayList<>());
+        samm.setMaterialGlobalAssetId(productItemStock.getMaterial().getMaterialNumberCx());
+        samm.setMaterialNumberSupplier(productItemStock.getMaterial().getOwnMaterialNumber());
+        samm.setMaterialNumberCustomer(mprService.find(productItemStock.getMaterial(),
+            productItemStock.getPartner()).getPartnerMaterialNumber());
+        var posList = new ArrayList<Position>();
+        samm.setPositions(posList);
+        Position position = new Position();
+        if (productItemStock.getCustomerOrderId() != null || productItemStock.getCustomerOrderPositionId() != null
+            || productItemStock.getSupplierOrderId() != null) {
+            OrderPositionReference opr = new OrderPositionReference(productItemStock.getSupplierOrderId(),
+                productItemStock.getCustomerOrderId(), productItemStock.getCustomerOrderPositionId());
+            position.setOrderPositionReference(opr);
+        }
+        ItemQuantityEntity itemQuantityEntity = new ItemQuantityEntity(productItemStock.getQuantity(),
+            productItemStock.getMeasurementUnit());
+        AllocatedStock allocatedStock = new AllocatedStock(itemQuantityEntity, productItemStock.getLocationBpns(),
+            productItemStock.isBlocked(), productItemStock.getLocationBpna());
+        var allocatedStocksList = new ArrayList<AllocatedStock>();
+        allocatedStocksList.add(allocatedStock);
+        position.setAllocatedStocks(allocatedStocksList);
+        return samm;
+    }
+
+    public List<ReportedProductItemStock> sammToReportedProductItemStock(ItemStockSAMM samm, Partner partner) {
+        String matNbrCustomer = samm.getMaterialNumberCustomer();
+        String matNbrSupplier = samm.getMaterialNumberSupplier(); // should be ownMaterialNumber
+        String matNbrCatenaX = samm.getMaterialGlobalAssetId();
+        ArrayList<ReportedProductItemStock> outputList = new ArrayList<>();
+        if(samm.getDirection() != DirectionCharacteristic.INBOUND) {
+            log.warn("Direction should be INBOUND, aborting");
+            return outputList;
+        }
+        Material material = null;
+        // Use CatenaXNbr
+        if (matNbrCatenaX != null) {
+            material = materialService.findByMaterialNumberCx(matNbrCatenaX);
+            if (material != null) {
+                if (!material.getOwnMaterialNumber().equals(matNbrSupplier)) {
+                    log.warn("Mismatch between CatenaX Number " + matNbrCatenaX + " and ownMaterialNumber " + matNbrSupplier);
+                }
+                var mpr = mprService.find(material, partner);
+                if (mpr == null) {
+                    log.warn("Missing MaterialPartnerRelation for " + material.getOwnMaterialNumber() + " and Partner " + partner.getBpnl());
+                } else {
+                    if (!mpr.getPartnerMaterialNumber().equals(matNbrCustomer)) {
```

Comment:
Similar to the other one. If I check the material numbers customer here instead (same as log) everything works in my test setup. But like this it doesn't.

Reply:
See above

Reply:
confirmed